crates/ruff_linter/src/rules/flake8_pyi/rules/non_self_return_type.rs

```diff
@@ -221,26 +221,14 @@ pub(crate) fn non_self_return_type(
 
 /// Returns `true` if the given class is a metaclass.
 fn is_metaclass(class_def: &ast::StmtClassDef, semantic: &SemanticModel) -> bool {
-    class_def.arguments.as_ref().is_some_and(|arguments| {
-        arguments
-            .args
-            .iter()
-            .any(|expr| is_metaclass_base(expr, semantic))
+    analyze::class::any_qualified_name(class_def, semantic, &|qualified_name| {
```

Comment:
IIRC this will recursively resolve subclasses (within the file). I'm not sure if that's intended in those case or not given the bullet in your PR summary.

Reply:
Superclasses rather than subclasses, right? ;)

But yup, that's what I want! Sorry if theh PR summary was unclear

Reply:
Uhh yes superclasses.

Reply:
Oh right, "The current logic" refers to `main`, not the current logic of _this_ PR. Of course.